.github/workflows/conda_unit_test.yml

```
@@ -1,24 +1,76 @@
 name: Unit Tests
 
-on: [push, pull_request]
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:   # allow manual triggering
+
+defaults:
+  run:
+    shell: bash -l {0}
 
 jobs:
 
+  lint:
+    name: Code style
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8
+
+      - name: Run flake8
+        run: |
+          flake8
+
+      - name: Run black
+        uses: rickstaa/action-black@v1
+        with:
+          black_args: ". --check"
+
   build:
 
     runs-on: ubuntu-latest
+    needs: lint
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9]
+        python-version:
+        - 3.8
+        - 3.9
+        - "3.10"
+        - 3.11
```

I'd convert all of them to strings for consistency.
```suggestion
        - "3.8"
        - "3.9"
        - "3.10"
        - "3.11"
```